components/ThreeDotsMenü/ThreeDotMenü.js

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from "react";
+import Link from "next/link";
+import FavoriteButton from "../FavoriteButton/FavoriteButton";
+import useLocalStorageState from "use-local-storage-state";
+import { StyledLink, StyledUL, StyledLI, StyledButton } from "./styles";
+
+function ThreeDotMenu({ article }) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [favorites, setFavorites] = useLocalStorageState("favorites", []);
+  const [isFavorite, setIsFavorite] = useState(false);
+
+  useEffect(() => {
+    if (!favorites) return;
+    const found = favorites.some(
+      (favorite) => favorite.shareURL === article.shareURL
+    );
+    setIsFavorite(found);
+  }, [favorites, article.shareURL]);
```

Comment:
The state of marked favorites should not be specific to the `ThreeDotMenu` component.
The component shouldn't care about the list of favorites. It should only be responsible for opening and closing its menu. This is called **separation of concerns.** The component shouldn't be concerned with managing the localStorage state.

Apart from that, there's another reason which is technical: The `ThreeDotsMenu` component is rendered once for every article you display. If you display 20 articles you render this component 20 times. And for every one of these 20 `ThreeDotMenus` you have a separate `state` and `useEffect`. But you only need to manage your state _once._

Lift up the state and the handle functions to a higher level and manage your state from there. Pass down your state to components that rely on it, like this component.